storage/validateArtifactLowerCase/validateArtifactLowerCase.groovy

```diff
@@ -0,0 +1,28 @@
+import org.artifactory.build.promotion.PromotionConfig
+import org.artifactory.build.staging.ModuleVersion
+import org.artifactory.build.staging.VcsConfig
+import org.artifactory.exception.CancelException
+import org.artifactory.repo.RepoPathFactory
+import org.artifactory.request.Request
+import org.artifactory.util.StringInputStream
+
+storage {
+    beforeCreate {
+        item - >
+            def repositorykey = item.repoPath.repoKey
+        def repositories = new JsonSlurper().parse(new File(ctx.artifactoryHome.haAwareEtcDir, 'plugins/validateArtifactLowerCase.json'))
```

Comment:
Instead of reading the configuration file for every request, it would be better to read it once during plugin load. You can do this by moving this line before the `storage` closure.

Reply:
This is debatable ... most plugins here read their config files each time they're called. It allows you to change the config and have those changes take effect immediately, without restarting Artifactory or reinstalling the plugin.

However, I do see your point. An extra disk read every time an artifact is requested might be expensive. I think I'm fine either way on this one.

Reply:
I believe that If changes are needed in the configuration, the user can call the Reload Plugin API instead of restarting Artifactory or reinstalling the plugin.

I agree that having configurations changes applied automatically is very helpful in most cases, but I believe that, for those events that are triggered for every request of a kind, we should try to minimize the performance impact the better we can.

Reply:
I agree with both the approach and feels like its better to check it once the plugin loads and not on every request.